doc: document INSIDE_DUNE in hacking.rst (#8221)
ocaml/dune: doc/hacking.rst

.. _ppx_expect:      https://github.com/janestreet/ppx_expect

When running dune inside tests, the ``INSIDE_DUNE`` environment variable is set.
This has the following effects:

* Change the default root detection behaviour to use the current directory
  rather than the top most ``dune-project`` / ``dune-workspace`` file.

> * Be less verbose various times Dune outputs a user message.

Reviewer: ```suggestion
* Be less verbose in the various times Dune outputs a user message.
```
…or “during the various times” ??
Feels like there’s something missing between “verbose” and “various”

Author: Hmm, sounds ok when I say it out loud, but I am no expert on grammar.

Author: ```suggestion
* Be less verbose when Dune outputs a user message.
```
tada this is better.

Reviewer: Yes! Perfect!

* Error reporting is deterministic by default.
* Prefer not to use a diff program for displaying diffs.

> This list is not exhaustive, and may change in the future. In order to find the

Reviewer: ```suggestion
This list is not exhaustive and may change in the future. In order to find the
```
